src/components/PriceLine.tsx

```diff
@@ -57,26 +58,51 @@ const PriceLine = ({
     if (!showBarrier) return null;
 
     return (
-        <div className='barrier-area' style={{ top: 0 }} ref={setDragLine} hidden={!visible}>
+        <div
+            className={classNames('barrier-area', { 'barrier-area--zero': isOverlappingWithPriceLine })}
+            style={{ top: 0 }}
+            ref={setDragLine}
+            hidden={!visible}
+        >
             <div
                 className={classNames('chart-line', 'horizontal', className || '', {
                     draggable,
                     dragging: isDragging,
                 })}
                 style={{
                     color: foregroundColor,
-                    backgroundImage:
-                        lineStyle && lineStyle !== 'solid' ? '' : `linear-gradient(to left, ${color} 90%, ${color}00`,
                 }}
             >
                 {showBarrierDragLine && (
-                    <div className='drag-line' style={{ borderTop: `${lineStyle} ${color} 1px` }} />
+                    <div
+                        className={classNames('drag-line', { 'drag-line--zero': isOverlappingWithPriceLine })}
+                        style={{
+                            borderTopStyle: lineStyle as React.CSSProperties['borderTopStyle'],
+                            width: `calc(100% - ${yAxiswidth}px + ${width}px)`,
+                        }}
+                    />
                 )}
                 <div className='draggable-area' />
-                <div className='drag-price' style={{ backgroundColor: color, width, opacity }}>
-                    <div className='price'>{priceDisplay}</div>
-                    {offScreen && offScreenDirection && (
-                        <PriceLineArrow offScreenDirection={offScreenDirection} color={color} />
+                <div className='draggable-area-wrapper'>
+                    <div
+                        className={classNames('drag-price')}
+                        style={{ width: isOverlappingWithPriceLine ? width : yAxiswidth, opacity, right: isOverlappingWithPriceLine ? yAxiswidth - width : 0 }}
+                    >
+                        <div className='drag-icon'>
+                            <div></div>
+                            <div></div>
+                            <div></div>
```

Comment:
👀

Reply:
why?

Reply:
This is to generate a hamburger menu icon without the use of a symbol as I want the icon to be separated from the value inside.

Reply:
Would it make sense to abstract it into a `HamburgerDragIcon` component so it's clear?

Reply:
i agree with @markwylde-deriv

Reply:
Well, it's just one place where we used it, so I don't see any difference except for having more lines of code, so I didn't add it. And we'll probably never need to reuse this component anywhere else, since it's charts as well. But I moved it into a separate component since everybody doesn't like to see this in the file 😄

Reply:
I think the difference is for documentation. The fact Bala had to ask what it's doing is, in my opinion, proof that it needs to be put into it's own component.

It doesn't necessarily need to go in it's own file.